pkg/cri/server/sandbox_stop.go

```diff
@@ -112,6 +112,11 @@ func (c *criService) stopPodSandbox(ctx context.Context, sandbox sandboxstore.Sa
 func (c *criService) stopSandboxContainer(ctx context.Context, sandbox sandboxstore.Sandbox) error {
 	id := sandbox.ID
 	container := sandbox.Container
+
+	if container == nil {
+		return cleanupUnknownSandbox(ctx, id, sandbox)
```

Comment:
@fuweid I am a little confused about this `cleanupUnknownSandbox`. It simply calls `handleSandboxExit`.  And `handleSandboxExit` ignores the passed event and does following things.

```
       err = sb.Status.Update(func(status sandboxstore.Status) (sandboxstore.Status, error) {
		status.State = sandboxstore.StateNotReady
		status.Pid = 0
		return status, nil
	})
	// Using channel to propagate the information of sandbox stop
	sb.Stop()
```

Is this what we want?

Reply:
In what case would this happen? I thought `Container` should be the final representation of the sandbox based on https://github.com/containerd/containerd/pull/5904#pullrequestreview-966050936?

Reply:
In current code, the only case is `sandbox.NetNS.Remove()` failure.

Reply:
In that case, we won't have the container object, only an in-memory state in the sandbox store. That isn't very useful to me.

If we don't have time to make the checkpoint sequence change, I would prefer to just keep the bug open, and fix it in a next PR. Only fixing it in the in-memory state would cause state mismatch. We consider the `container` as the source of truth of whether the sandbox exists before, now this is changing the source of truth to the in-memory state.

Reply:
Talked offline, this code block should be unnecessary. We can prevent nil container from happening in the first place.